src/Collection/CollectionTrait.php

```diff
@@ -702,7 +702,7 @@ public function isEmpty()
     public function unwrap()
     {
         $iterator = $this;
-        while (get_class($iterator) === 'Cake\Collection\Collection') {
+        while ($iterator instanceof Collection) {
```

Comment:
Can we have an inline comment about such decisions that are not apparent from the code bae itself?